subsys/net/ip/net_core.c

```diff
@@ -28,6 +28,9 @@ LOG_MODULE_REGISTER(net_core, CONFIG_NET_CORE_LOG_LEVEL);
 #include <net/dns_resolve.h>
 #include <net/gptp.h>
 #include <net/websocket.h>
+#if defined(CONFIG_NET_DSA)
```

Comment:
We probably do not need to have ifdef here.

Reply:
The #ifdef could be removed if the dsa_recv_set_iface() would have an empty implementation when CONFIG_NET_DSA is not defined.